plutus-tx/src/PlutusTx/AssocMap.hs

```diff
@@ -70,8 +71,20 @@ import Prettyprinter (Pretty (..))
 -- Take care when using 'fromBuiltinData' and 'unsafeFromBuiltinData', as neither function performs
 -- deduplication of the input collection and may create invalid 'Map's!
 newtype Map k v = Map {unMap :: [(k, v)]}
-  deriving stock (Generic, Haskell.Eq, Haskell.Show, Data, TH.Lift)
-  deriving newtype (Eq, Ord, NFData)
+  deriving stock (Generic, Haskell.Show, Data, TH.Lift)
+  deriving newtype (NFData)
+
+-- | Equality check for `Map`. Warning: this operation is O(n^2).
+instance (Eq k, Eq v) => Eq (Map k v) where
```

Comment:
Maybe we should reconsider using SortedMap? That one is O(n)  https://github.com/IntersectMBO/plutus/pull/5812